### ISSUE TITLE
Test environment can't be setup

### ISSUE THREAD

**Issue by dagoroz** [Azure/azure-powershell#18425]
### Description

I tried setting up my machine for working on my team's module and followed the set-up [here ](https://github.com/Azure/azure-powershell/blob/main/documentation/testing-docs/using-azure-test-framework.md) and I still cannot run the tests.

everything "works" fine until the actual test execution (see debug output)

### Issue script & Debug output

```PowerShell
Microsoft.Azure.Commands.Resources.Test.ScenarioTests.RoleAssignmentTests.RaDeleteByPSRoleAssignment
   Source: RoleAssignmentTests.cs line 78
   Duration: 4.7 sec

  Message: 
System.Management.Automation.ActionPreferenceStopException : The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: CompactToken parsing failed with error code: 80049217

  Stack Trace: 
PipelineBase.Invoke(IEnumerable input)
Pipeline.Invoke()
Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
PowerShell.Invoke()
EnvironmentSetupHelper.RunPowerShellTest(String[] scripts) line 575
TestManager.RunTestScript(String[] scripts) line 227
RoleAssignmentTests.RaDeleteByPSRoleAssignment() line 80

  Standard Output: 
Test-RaDeleteByPSRoleAssignment
Caught Exception: System.Management.Automation.ActionPreferenceStopException: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: CompactToken parsing failed with error code: 80049217
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
   at System.Management.Automation.Runspaces.Pipeline.Invoke()
   at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
   at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
   at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
   at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
   at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
   at System.Management.Automation.PowerShell.Invoke()
   at Microsoft.WindowsAzure.Commands.ScenarioTest.EnvironmentSetupHelper.RunPowerShellTest(String[] scripts) in C:\src\azure-powershell\tools\ScenarioTest.ResourceManager\EnvironmentSetupHelper.cs:line 575
Message: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: CompactToken parsing failed with error code: 80049217
PowerShell Error Record: CompactToken parsing failed with error code: 80049217
Exception:System.Exception: [InvalidAuthenticationToken] : CompactToken parsing failed with error code: 80049217
Details:CompactToken parsing failed with error code: 80049217
Script Stack Trace: at Get-AzADUser<Process>, C:\src\azure-powershell\artifacts\Debug\Az.Resources\MSGraph.Autorest\internal\ProxyCmdletDefinitions.ps1: line 2197
at Get-AzADUser<Process>, C:\src\azure-powershell\artifacts\Debug\Az.Resources\MSGraph.Autorest\custom\Get-AzADUser.ps1: line 199
at Get-AzADUser<Process>, C:\src\azure-powershell\artifacts\Debug\Az.Resources\MSGraph.Autorest\exports\ProxyCmdletDefinitions.ps1: line 3604
at Test-RaDeleteByPSRoleAssignment, C:\src\azure-powershell\src\Resources\Resources.Test\bin\Debug\netcoreapp3.1\ScenarioTests\RoleAssignmentTests.ps1: line 95
at <ScriptBlock>, <No file>: line 1
: Target: { ConsistencyLevel = , Search = , Filter = , Count = , Orderby = , Select = , Expand =  }

================== COMMANDS =======================

Set-ExecutionPolicy Unrestricted -Scope Process -ErrorAction Ignore

$error.clear()

Write-Debug "current directory: C:\src\azure-powershell\src\Resources\Resources.Test\bin\Debug\netcoreapp3.1\"

Write-Debug "current executing assembly: C:\src\azure-powershell\src\Resources\Resources.Test\bin\Debug\netcoreapp3.1"

cd "C:\src\azure-powershell\src\Resources\Resources.Test\bin\Debug\netcoreapp3.1\"

Import-Module "C:\src\azure-powershell\src\Resources\Resources.Test\bin\Debug\netcoreapp3.1\Assert.ps1"

Import-Module "C:\src\azure-powershell\src\Resources\Resources.Test\bin\Debug\netcoreapp3.1\Common.ps1"

Import-Module "C:\src\azure-powershell\artifacts\Debug\Az.Accounts\Az.Accounts.psd1"

Enable-AzureRmAlias -Module Az.Accounts

Import-Module "C:\src\azure-powershell\artifacts\Debug\Az.Resources\Az.Resources.psd1"

Enable-AzureRmAlias -Module Az.Resources

Import-Module "C:\src\azure-powershell\artifacts\Debug\Az.Resources\Az.Resources.psd1"

Enable-AzureRmAlias -Module Az.Resources

Import-Module "C:\src\azure-powershell\artifacts\Debug\Az.Monitor\Az.Monitor.psd1"

Enable-AzureRmAlias -Module Az.Monitor

Import-Module "C:\src\azure-powershell\artifacts\Debug\Az.ManagedServiceIdentity\Az.ManagedServiceIdentity.psd1"

Enable-AzureRmAlias -Module Az.ManagedServiceIdentity

Import-Module "C:\src\azure-powershell\src\Resources\Resources.Test\bin\Debug\netcoreapp3.1\ScenarioTests\Common.ps1"

Import-Module "C:\src\azure-powershell\src\Resources\Resources.Test\bin\Debug\netcoreapp3.1\ScenarioTests\RoleAssignmentTests.ps1"

Disable-AzureRmDataCollection -ErrorAction Ignore

set-location "C:\src\azure-powershell\src\Resources\Resources.Test\bin\Debug\netcoreapp3.1\"

$TestOutputRoot='C:\src\azure-powershell\src\Resources\Resources.Test\bin\Debug\netcoreapp3.1\'

$VerbosePreference='Continue'

$DebugPreference='Continue'

$ErrorActionPreference='Stop'

Write-Debug "AZURE_TEST_MODE = $($env:AZURE_TEST_MODE)"

Write-Debug "TEST_HTTPMOCK_OUTPUT =  $($env:TEST_HTTPMOCK_OUTPUT)"

Test-RaDeleteByPSRoleAssignment

===================================================

---------------------------------------------------------------

DEBUG STREAM

---------------------------------------------------------------

AZURE_TEST_MODE = Playback

TEST_HTTPMOCK_OUTPUT =  

[CmdletBeginProcessing]: Starting command

CmdletBeginProcessing: 

CmdletProcessRecordStart: 

Client side pagination is enabled for this cmdlet

CmdletGetPipeline: 

CmdletBeforeAPICall: 

URLCreated: /users

RequestCreated: /v1.0/users

HeaderParametersAdded: 

============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://graph.microsoft.com/v1.0/users

Headers:
x-ms-unique-id                : 1
x-ms-client-request-id        : 8ee0ed62-0ea6-42f6-b861-9e2ba49fd58f
CommandName                   : Az.MSGraph.internal\Get-AzADUser
FullCommandName               : Get-AzADUser_List
ParameterSetName              : __AllParameterSets
User-Agent                    : AzurePowershell/v8.0.0,PSVersion/v3.0.0.0,Az.MSGraph/6.0.0

Body:




BeforeCall: 

============================ HTTP RESPONSE ============================

Status Code:
Unauthorized

Headers:
Transfer-Encoding             : chunked
Strict-Transport-Security     : max-age=31536000
request-id                    : 131f9d2d-dd08-427c-8634-68d4afec6262
client-request-id             : 131f9d2d-dd08-427c-8634-68d4afec6262
x-ms-ags-diagnostic           : {"ServerInfo":{"DataCenter":"West US","Slice":"E","Ring":"4","ScaleUnit":"000","RoleInstance":"BY1PEPF00002043"}}
WWW-Authenticate              : Bearer realm="", authorization_uri="https://login.microsoftonline.com/common/oauth2/authorize", client_id="00000003-0000-0000-c000-000000000000"
Date                          : Tue, 07 Jun 2022 19:49:30 GMT

Body:
{
  "error": {
    "code": "InvalidAuthenticationToken",
    "message": "CompactToken parsing failed with error code: 80049217",
    "innerError": {
      "date": "2022-06-07T19:49:31",
      "request-id": "131f9d2d-dd08-427c-8634-68d4afec6262",
      "client-request-id": "131f9d2d-dd08-427c-8634-68d4afec6262"
    }
  }
}



ResponseCreated: 

BeforeResponseDispatch: 

CmdletProcessRecordEnd: 

---------------------------------------------------------------
```

### Environment data

```PowerShell
issue with source code not PS itself
```

### Module versions

```PowerShell
issue with source code not PS itself
```

### Error output

```PowerShell
issue with source code not PS itself
```

**Comment by dingmeng-xue** [comment]
@vidai-msft please help to look into it.

**Comment by vidai-msft** [comment]
@dagoroz This is a known limitation. Currently our test framework only supports pure management plane cmdlets but not for the others like data plane cmdlets and autorest cmdlets. This test ps1 has mixed management plane and autorest cmdlets which resulted in the exception. You can filter out some test cases using Trait:"AcceptanceType [Checkin]" in Test Explorer.

**Comment by dagoroz** [comment]
@vidai-msft thank you very much, I ran a chekin test and worked perfectly.
I have a couple of followup questions.

1. does this limitation extend to record mode or is it only a playback thing?
2. is there some timeline to address this limitation? is it even something that can be addressed?

**Comment by vidai-msft** [comment]
@dagoroz please see my comment below.
1. Yes, the limitation extends to record mode also because currently both record and playback mode relies on the mock context which is not compatible with autorest.
2. Actually we are planning to improve the existing testfx. We will evaluate the implementation of the autorest cmdlets and see if it is possible to address the limitation. But still has lots of work to do. The rough ETA should be by the end of this CY or next year.

**Comment by dagoroz** [comment]
@vidai-msft Excelent
I have a last couple of questions regarding development and testing.

1. I managed to start a test in record mode and the error changed from what I pasted in this ticket to "auth token audience is wrong" are this 2 problems related or should I raise a new bug report?
2. considering the test infra does not allow to test nor debug locally is there any guidance on how to repro issues and/or test changes do what we want? To give a more specific example the Remove-AzRoleAssignment cmdlet is not giving feedback when executing successfully and I was supposed to fix it but it's impossible if I cannot debug the code.
